bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java

```diff
@@ -0,0 +1,371 @@
+import static org.jboss.as.controller.client.helpers.ClientConstants.READ_ATTRIBUTE_OPERATION;
+import static org.jboss.as.controller.client.helpers.ClientConstants.RESULT;
+import static org.jboss.as.controller.client.helpers.ClientConstants.RUNTIME_NAME;
+import org.jboss.as.process.CommandLineConstants;
+import org.jboss.as.process.ExitCodes;
+import org.jboss.dmr.ModelNode;
+import org.jboss.logmanager.LogContext;
+import org.jboss.logmanager.PropertyConfigurator;
+import org.jboss.modules.ModuleClassLoader;
+import org.jboss.modules.ModuleLoader;
+import static org.wildfly.core.jar.runtime.Constants.JBOSS_SERVER_CONFIG_DIR;
+import static org.wildfly.core.jar.runtime.Constants.JBOSS_SERVER_LOG_DIR;
+import static org.wildfly.core.jar.runtime.Constants.LOG_BOOT_FILE_PROP;
+import static org.wildfly.core.jar.runtime.Constants.LOG_MANAGER_CLASS;
+import static org.wildfly.core.jar.runtime.Constants.LOG_MANAGER_PROP;
+import static org.wildfly.core.jar.runtime.Constants.STANDALONE;
+import static org.wildfly.core.jar.runtime.Constants.STANDALONE_CONFIG;
+import org.wildfly.core.jar.runtime._private.BootableJarLogger;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import static org.wildfly.core.jar.runtime.Constants.CONFIGURATION;
+import static org.wildfly.core.jar.runtime.Constants.DATA;
+import static org.wildfly.core.jar.runtime.Constants.DEPLOYMENTS;
+import static org.wildfly.core.jar.runtime.Constants.LOG;
+import static org.wildfly.core.jar.runtime.Constants.LOGGING_PROPERTIES;
+import static org.wildfly.core.jar.runtime.Constants.SERVER_LOG;
+import static org.wildfly.core.jar.runtime.Constants.SERVER_STATE;
+import static org.wildfly.core.jar.runtime.Constants.SHA1;
+import static org.wildfly.core.jar.runtime.Constants.STOPPED;
+import org.wildfly.core.jar.runtime.Server.ShutdownHandler;
+
+/**
+ *
+ * @author jdenise
+ */
+public final class BootableJar implements ShutdownHandler {
+
+    private static final String DEP_1 = "ff";
+    private static final String DEP_2 = "00";
+
+    private class ShutdownHook extends Thread {
+
+        @Override
+        public void run() {
+            log.shuttingDown();
+            waitAndClean();
+        }
+    }
+
+    private BootableJarLogger log;
+
+    private final Path jbossHome;
+    private final List<String> startServerArgs = new ArrayList<>();
+    private Server server;
+    private final Arguments arguments;
+    private final ModuleLoader loader;
+
+    private BootableJar(Path jbossHome, Arguments arguments, ModuleLoader loader, long unzipTime) throws Exception {
+        this.jbossHome = jbossHome;
+        this.arguments = arguments;
+        this.loader = loader;
+        startServerArgs.addAll(arguments.getServerArguments());
+        startServerArgs.add(CommandLineConstants.READ_ONLY_SERVER_CONFIG + "=" + STANDALONE_CONFIG);
+
+        configureLogger();
```

Comment:
This is just a minor note. It might be useful to add a comment here that logging needs to be configured before other components have a chance to initialize a logger.

Reply:
Ok, added a comment.